### PR TITLE
python from the shell slides

### LINKED ISSUE #21: Slides: Python from the Shell
We don't seem to have much prior art to draw on here.

@bradleyboehmke / @eswan18 / @augustopher, what kind of topics did you have in mind here? Possibilities:

* Running Python from the command line
* Creating virtual environments
* The subprocess module for running scripts from Python
* Using argparse to parse command line arguments

### PR DESCRIPTION
**Pull request by cunningjames**
Closes #21.

This is a fairly straight port of Ethan's slides from the intermediate class, switched from Jupyter to VS Code. It's good material to have, though I wonder if some of the class will find it a bit basic. I'm happy to revisit if you guys think it's necessary.

From a technical perspective, I'm not sure whether or not they'll be able to use ipython easily, particularly the Windows users, but we can deal with that as necessary.

### CODE REVIEW

File: notebooks/12-Python_from_the_Shell.ipynb [hunk 1]
```diff
@@ -0,0 +1,1117 @@
...
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "<font class=\"your_turn\">\n",
+    "Your Turn\n",
+    "</font>\n",
```
Comment (augustopher) [inline notebooks/12-Python_from_the_Shell.ipynb]: Update to a `h3` header.

File: notebooks/12-Python_from_the_Shell.ipynb [hunk 2]
```diff
@@ -0,0 +1,1117 @@
...
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "source": [
+    "<div class=\"question\">\n",
+    "<strong>Question:</strong> What is the file extension for Jupyter notebooks? Why is this different from .py?\n",
+    "</div>"
```
Comment (augustopher) [inline notebooks/12-Python_from_the_Shell.ipynb]: Remove the custom formatting and just bold it.

File: notebooks/12-Python_from_the_Shell.ipynb [hunk 3]
```diff
@@ -0,0 +1,1117 @@
...
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "<font class=\"your_turn\">\n",
+    "Your Turn\n",
+    "</font>\n",
```
Comment (augustopher) [inline notebooks/12-Python_from_the_Shell.ipynb]: Convert to a `h3` header.

File: notebooks/12-Python_from_the_Shell.ipynb [hunk 4]
```diff
@@ -0,0 +1,1117 @@
...
+    "\n",
+    "Open a new text file in Jupyter and type the same code as before:\n",
```
Comment (augustopher) [inline notebooks/12-Python_from_the_Shell.ipynb]:
```suggestion
    "Open a new text file in VS Code and type the same code as before:\n",
```

File: notebooks/12-Python_from_the_Shell.ipynb [hunk 5]
```diff
@@ -0,0 +1,1117 @@
...
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "<font class=\"your_turn\">\n",
+    "Your Turn\n",
+    "</font>\n",
```
Comment (augustopher) [inline notebooks/12-Python_from_the_Shell.ipynb]: Convert to `h3` header.

File: notebooks/12-Python_from_the_Shell.ipynb [hunk 6]
```diff
@@ -0,0 +1,1117 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "#### Graphics"
```
Comment (augustopher) [inline notebooks/12-Python_from_the_Shell.ipynb]:
```suggestion
    "### Graphics"
```

File: notebooks/12-Python_from_the_Shell.ipynb [hunk 7]
```diff
@@ -0,0 +1,1117 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "#### Automatic Printing of Output"
```
Comment (augustopher) [inline notebooks/12-Python_from_the_Shell.ipynb]:
```suggestion
    "### Automatic Printing of Output"
```

File: notebooks/12-Python_from_the_Shell.ipynb [hunk 8]
```diff
@@ -0,0 +1,1117 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "#### Cell Magics and `?`"
```
Comment (augustopher) [inline notebooks/12-Python_from_the_Shell.ipynb]:
```suggestion
    "### Cell Magics and `?`"
```

File: notebooks/12-Python_from_the_Shell.ipynb [hunk 9]
```diff
@@ -0,0 +1,1117 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
```
Comment (augustopher) [inline notebooks/12-Python_from_the_Shell.ipynb]: To match the flow of this section, this should build into the title from above:

```suggestion
     "slide_type": "fragment"
```

File: notebooks/12-Python_from_the_Shell.ipynb [hunk 10]
```diff
@@ -0,0 +1,1117 @@
...
+    }
+   },
+   "source": [
+    "#### Batching Code"
```
Comment (augustopher) [inline notebooks/12-Python_from_the_Shell.ipynb]:
```suggestion
    "### Batching Code"
```

**Review by augustopher — CHANGES_REQUESTED** [review]
Also need to add the usual "Questions" slide at the end.

**Comment by cunningjames** [comment]
Thanks, @augustopher. I think I've addressed all your comments.

File: notebooks/12-Python_from_the_Shell.ipynb [hunk 11]
```diff
@@ -0,0 +1,1124 @@
...
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "<font class=\"your_turn\">\n",
+    "Your Turn\n",
+    "</font>\n",
```
Comment (augustopher) [inline notebooks/12-Python_from_the_Shell.ipynb]: Guess I missed one here.

Comment (cunningjames) [reply notebooks/12-Python_from_the_Shell.ipynb]: Fixed, thanks!

**Review by augustopher — APPROVED** [review]

**Comment by eswan18** [comment]
FYI – feel free to merge these with just one approval. I have work events most of the day tomorrow and into the evening, and then again the next day, so I won't be able to review all these.

I'll do a once-over of all of the material later in the week.

**Pull request merged**